src/worker/jobs/downloads.rs

```diff
@@ -0,0 +1,521 @@
+            .collect::<Vec<_>>();
+
+        info!("Top 30 downloads: {top_downloads:?}");
+
+        Ok(())
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize, clap::Parser)]
+pub struct ProcessCdnLogQueue {
+    /// The maximum number of messages to receive from the queue and process.
+    #[clap(long, default_value = "1")]
+    max_messages: usize,
+}
+
+impl BackgroundJob for ProcessCdnLogQueue {
+    const JOB_NAME: &'static str = "process_cdn_log_queue";
+
+    type Context = Arc<Environment>;
+
+    async fn run(&self, ctx: Self::Context) -> anyhow::Result<()> {
+        let queue = Self::build_queue(&ctx.config.cdn_log_queue);
+        self.run(queue, &ctx.connection_pool).await
+    }
+}
+
+impl ProcessCdnLogQueue {
+    fn build_queue(config: &CdnLogQueueConfig) -> Box<dyn SqsQueue + Send + Sync> {
+        match config {
+            CdnLogQueueConfig::Mock => Box::new(MockSqsQueue::new()),
+            CdnLogQueueConfig::SQS {
+                access_key,
+                secret_key,
+                region,
+                queue_url,
+            } => {
+                use secrecy::ExposeSecret;
+
+                let secret_key = secret_key.expose_secret();
+                let credentials = Credentials::from_keys(access_key, secret_key, None);
+
+                let region = Region::new(region.to_owned());
+
+                Box::new(SqsQueueImpl::new(queue_url, region, credentials))
+            }
+        }
+    }
+
+    async fn run(
+        &self,
+        queue: Box<dyn SqsQueue + Send + Sync>,
+        connection_pool: &DieselPool,
+    ) -> anyhow::Result<()> {
+        const MAX_BATCH_SIZE: usize = 10;
+
+        info!("Receiving messages from the CDN log queue…");
+        let mut num_remaining = self.max_messages;
+        while num_remaining > 0 {
+            let batch_size = num_remaining.min(MAX_BATCH_SIZE);
+            num_remaining -= batch_size;
+
+            debug!("Receiving next {batch_size} messages from the CDN log queue…");
+            let response = queue.receive_messages(batch_size as i32).await?;
+
+            let messages = response.messages();
+            debug!(
+                "Received {num_messages} messages from the CDN log queue",
+                num_messages = messages.len()
+            );
+            if messages.is_empty() {
+                info!("No more messages to receive from the CDN log queue");
+                break;
+            }
+
+            for message in messages {
+                let message_id = message.message_id().unwrap_or("<unknown>");
+                debug!("Processing message: {message_id}");
+
+                let Some(receipt_handle) = message.receipt_handle() else {
+                    warn!("Message {message_id} has no receipt handle; skipping");
+                    continue;
+                };
+
+                debug!("Deleting message {message_id} from the CDN log queue…");
+                queue
+                    .delete_message(receipt_handle)
```

Comment:
Maybe a question for infra, but this queue is configured to only deliver messages exactly once, right?

(What I'm concerned about here is double counting if this job runs twice, and the second `receive_messages()` call occurs before messages being handled by the first job are deleted.)

Reply:
> this queue is configured to only deliver messages exactly once, right?

from what I've seen so far that does not appear to be the case. I assume if it was the case then the delivered messages would be auto-deleted upon reception, right?

Reply:
I will treat this as a non-blocking concern since the code here is experimental and read-only for now anyway. If we can change the queue to use once-only delivery then we can just delete the `delete_message()` calls here.

Reply:
> Maybe a question for infra, but this queue is configured to only deliver messages exactly once, right?

This is a [standard queue](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/standard-queues.html) that only guarantees [`at least once`](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/standard-queues-at-least-once-delivery.html) delivery. If we want [`exactly once`](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/FIFO-queues-exactly-once-processing.html), we'd need to recreate the queue as a [FIFO queue](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-fifo-queues.html).